sourced/ml/cmd_entries/id_splitter.py

```diff
@@ -0,0 +1,111 @@
+import argparse
+import string
+
+from sourced.ml.cmd_entries import ArgumentDefaultsHelpFormatterNoNone
+from sourced.ml.algorithms.id_splitter.nn_model import prepare_rnn_model, prepare_cnn_model, \
+    DEFAULT_RNN_TYPE
+from sourced.ml.algorithms.id_splitter.pipeline import pipeline
+from sourced.ml.utils.engine import pause
+
+# Article parameters
+# Common parameters
+MAXLEN = 40  # max length of sequence
+PADDING = "post"  # add padding values after input
+EPOCHS = 10
+BATCH_SIZE = 500
+VAL_BATCH_SIZE = 2000
+START_LR = 0.001
+FINAL_LR = 0.00001
+DEFAULT_DEVICES = "0"
+RANDOM_SEED = 1989
+SAMPLES_BEFORE_REPORT = 5 * 10 ** 6
+TEST_SIZE = 0.2  # fraction of dataset to use as test
+
+# CSV default parameters
+TOKEN_COL = 3
+TOKEN_SPLIT_COL = 4
+
+# RNN default parameters
+RNN_TYPES = ("GRU", "LSTM", "CuDNNLSTM", "CuDNNGRU")
+
+# CNN default parameters
+FILTERS = "64,32,16,8"
+KERNEL_SIZES = "2,4,8,16"
+DIM_REDUCTION = 32
+
+
+@pause
+def rnn_pipeline(args):
+    return pipeline(args, prepare_model=prepare_rnn_model)
+
+
+@pause
+def cnn_pipeline(args):
+    return pipeline(args, prepare_model=prepare_cnn_model)
+
+
+def add_id_splitter_arguments(parser: argparse.ArgumentParser):
+    subparsers = parser.add_subparsers(help="Identifier splitter", dest="id-splitter")
+
+    def add_parser(name, help_message):
+        return subparsers.add_parser(
+            name, help=help_message, formatter_class=ArgumentDefaultsHelpFormatterNoNone)
+    # common arguments for CNN/RNN models
+    parser.add_argument("-i", "--input", help="Path to CSV file.", required=True)
+    parser.add_argument("-e", "--epochs", type=int, default=EPOCHS, help="Number of epochs.")
```

Comment:
Number of training epochs. The more the better but the training time is proportional.